### PR TITLE
Optimize random turns in dungeongen

### PR DESCRIPTION
**Pull request by osjc**
Here are some unnecessary-string-by-value and unnecessary-temporaries fixes. One turned out to be rather peculiar ("optimize animations in client" where the code relied on the parameter passed by value for correctness, the code was updated to use the already made copy of that value in a class member) but the rest is, according to my humble opinion, rather straightforward.

This is a rather peculiar fix that according to my opinion merits its own pull request so the community has a chance to check whether the proposed fix is going to work, especially in long term.

### CODE REVIEW

File: src/mapgen/dungeongen.cpp [hunk 1]
```diff
@@ -651,20 +651,17 @@ v3s16 turn_xz(v3s16 olddir, int t)
 }
 
 
-v3s16 random_turn(PseudoRandom &random, v3s16 olddir)
+void random_turn(PseudoRandom &random, v3s16 &olddir)
```
Comment (paramat) [inline src/mapgen/dungeongen.cpp]: 'olddir' seems wrong now, perhaps 'dir'?

File: src/mapgen/dungeongen.cpp [hunk 2]
```diff
@@ -651,20 +651,17 @@ v3s16 turn_xz(v3s16 olddir, int t)
...
 {
 	int turn = random.range(0, 2);
-	v3s16 dir;
-	if (turn == 0)
-		// Go straight
-		dir = olddir;
-	else if (turn == 1)
+	if (turn == 0) {
+		// Go straight: nothing to do
```
Comment (nerzhul) [inline src/mapgen/dungeongen.cpp]: please replace the noop with a return

File: src/mapgen/dungeongen.cpp [hunk 3]
```diff
@@ -651,20 +651,18 @@ v3s16 turn_xz(v3s16 olddir, int t)
...
+		return;
+	} else if (turn == 1)
 		// Turn right
-		dir = turn_xz(olddir, 0);
+		olddir = turn_xz(olddir, 0);
```
Comment (paramat) [inline src/mapgen/dungeongen.cpp]: dir =

File: src/mapgen/dungeongen.cpp [hunk 4]
```diff
@@ -651,20 +651,18 @@ v3s16 turn_xz(v3s16 olddir, int t)
...
 	else
 		// Turn left
-		dir = turn_xz(olddir, 1);
-	return dir;
+		olddir = turn_xz(olddir, 1);
```
Comment (paramat) [inline src/mapgen/dungeongen.cpp]: dir =
?

File: src/mapgen/dungeongen.cpp (hunk 1 above)
Comment (osjc) [reply src/mapgen/dungeongen.cpp]: Resolved by SmallJoker

**Comment by osjc** [comment]
I like the new version made by SmallJoker.

File: src/mapgen/dungeongen.cpp [hunk 5]
```diff
@@ -651,20 +651,18 @@ v3s16 turn_xz(v3s16 olddir, int t)
...
-v3s16 random_turn(PseudoRandom &random, v3s16 olddir)
+void random_turn(PseudoRandom &random, v3s16 &dir)
 {
 	int turn = random.range(0, 2);
-	v3s16 dir;
-	if (turn == 0)
-		// Go straight
-		dir = olddir;
-	else if (turn == 1)
+	if (turn == 0) {
```
Comment (paramat) [inline src/mapgen/dungeongen.cpp]: Bracket shouldn't be used as it isn't used below.

File: src/mapgen/dungeongen.cpp [hunk 6]
```diff
@@ -651,20 +651,18 @@ v3s16 turn_xz(v3s16 olddir, int t)
...
+		// Go straight: nothing to do
+		return;
+	} else if (turn == 1)
```
Comment (paramat) [inline src/mapgen/dungeongen.cpp]: Bracket shouldn't be used as it isn't used below.

File: src/mapgen/dungeongen.cpp [hunk 7]
```diff
@@ -651,20 +651,18 @@ v3s16 turn_xz(v3s16 olddir, int t)
...
 		// Turn right
 		dir = turn_xz(olddir, 0);
```
Comment (paramat) [inline src/mapgen/dungeongen.cpp]: 'olddir' isn't used anymore.

File: src/mapgen/dungeongen.cpp [hunk 8]
```diff
@@ -651,20 +651,18 @@ v3s16 turn_xz(v3s16 olddir, int t)
...
 	else
 		// Turn left
 		dir = turn_xz(olddir, 1);
```
Comment (paramat) [inline src/mapgen/dungeongen.cpp]: 'olddir' isn't used anymore.

**Comment by paramat** [comment]
Errros need fixing.

**Comment by paramat** [comment]
:+1:

**Pull request merged**